ouroboros-consensus-test/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -650,12 +679,30 @@ instance Arbitrary TestSetupWithTxInMempool where
     return $ TestSetupWithTxInMempool testSetup tx
 
   shrink (TestSetupWithTxInMempool testSetup _tx) =
-    [ TestSetupWithTxInMempool testSetup tx'
+    [ TestSetupWithTxInMempool testSetup' tx'
     | testSetup' <- shrink testSetup
     , not . null . testInitialTxs $ testSetup'
     , tx' <- testInitialTxs testSetup'
     ]
 
+data TestSetupWithTxsInMempool = TestSetupWithTxsInMempool TestSetup [TestTx]
+  deriving (Show)
+
+instance Arbitrary TestSetupWithTxsInMempool where
+  arbitrary = do
+    TestSetupWithTxs { testSetup } <-
+      arbitrary `suchThat` (not . null . testInitialTxs . testSetup)
+    txs <- sublistOf (testInitialTxs testSetup)
+    return $ TestSetupWithTxsInMempool testSetup txs
+
+  -- shrink (TestSetupWithTxInMempool testSetup _tx) =
```

Comment:
Let's go with either just `TODO shrink` or fix it up now (eg if you just forgot to).

Reply:
Will put a TODO for now.